distribution/metadata/blobsum_service.go

```diff
@@ -13,8 +13,14 @@ type BlobSumService struct {
 	store Store
 }
 
+// BlobSum contains the digest and source repository information for a layer.
+type BlobSum struct {
```

Comment:
Thinking about this a little more, this should really be a reference type. We have two styles of references this could take. In one case, it can be just a digest reference and the other case can be a canonical reference. The canonical case covers the case we where insert a repository name into the metadata store.

Another option here, which is probably better, is to take options to all these methods that annotate the source repositories. It would require a second call into a constituent service but it would be a lot cleaner.